import/controls/permissionmanager.cpp

```diff
@@ -26,11 +29,13 @@ void PermissionManager::add(const Permission &permission)
                 permission.m_type,
                 permission.m_capability,
                 permission.m_expireType);
+    sendRequest(QStringLiteral("get-all"));
```

Comment:
Rather than reading all back you should consider adding notify to the embedlite after successfully added. Similarly for removed case.

Bare minimum would be that this would call, "sendRequest(QStringLiteral("get-all-for-uri"), permission.m_host)";

See... PermissionModel::requestPermissions(const QString &host)

Reply:
On embedlite-components side and in ContentPermissionManager we could handle "perm-changed" and build and send updated list after changes are made.

You can find a browser component usage from here https://github.com/sailfishos/gecko-dev/blob/master/browser/components/preferences/sitePermissions.js#L97

So adding Services.obs.addObserver(this, "perm-changed") upon start and clearing it upon shutdown  Services.obs.removeObserver(this, "perm-changed").